org.uqbar.project.wollok/src/org/uqbar/project/wollok/model/WollokModelExtensions.xtend

```diff
@@ -435,4 +466,35 @@ class WollokModelExtensions {
 		val rootNodeName = e.name.trim
 		node.text.trim.equals(rootNodeName) && semanticsElements.contains(node.semanticElement.eClass.name) && doApplyRenameTo(e, node.semanticElement)
 	}
+	
+	def static dispatch expectsExpression(EObject e) { false }
```

Comment:
For (a) well, we should add more tests to cover more situations, (b) I didn't catch you, maybe an example could lead us to a test and enhance this method.